content/kubeone/master/using_kubeone/manual_cluster_recovery.en.md

```diff
@@ -0,0 +1,352 @@
+name.
+
+In case of AWS, the following taint commands should be used:
+
+```bash
+terraform taint 'aws_instance.control_plane[0]'
+terraform taint 'aws_instance.control_plane[1]'
+terraform taint 'aws_instance.control_plane[2]'
+```
+
+Running the `apply` command will recreate the instances and update the load
+balancer to point to the new instances.
+
+```bash
+terraform apply
+```
+
+If needed, create the new Terraform output file:
+
+```bash
+terraform output -json > tf.json
+```
+
+#### Recreating Instances Manually
+
+If you are managing the infrastructure manually, you need to remove and create
+instances using your preferred method for managing infrastructure. Once the
+new instances are created, update the KubeOne configuration manifest.
+
+The information about the instances are located in the `hosts` part of the
+configuration manifest:
+
+```yaml
+apiVersion: kubeone.io/v1alpha1
+kind: KubeOneCluster
+name: demo-cluster
+hosts:
+- privateAddress: '172.18.0.1'
+  ...
+  isLeader: true
+- privateAddress: '172.18.0.2'
+  ...
+- privateAddress: '172.18.0.3'
+  ...
+```
+
+### Option 2: Reset The Cluster Using KubeOne
+
+{{% notice warning %}}
+This is not recommended because if something is broken on the instance itself,
+it can affect the newly created cluster as well.
+{{% /notice %}}
+
+If you're not able to recreate the VM instances you can reuse the existing
+ones.
+
+Unprovision the cluster by running the `reset` command such as:
+
+```bash
+kubeone reset config.yaml -t . --destroy-workers=false
+```
+
+After this is done, ensure that the `/etc/kubernetes` directory is empty on all
+control-plane instances. You can do that by SSH-ing to each instance and 
+running:
+
+```bash
+sudo rm -rf /etc/kubernetes/*
+```
+
+## Step 3 — Install The Kubernetes Binaries
+
+Once you have the new instances, you need to install the Kubernetes binaries
+before you restore the backup. We'll not provision the new cluster at this
+stage because we want to restore all the needed files first.
+
+Run the following command to install the prerequisites and the Kubernetes
+binaries:
+
+```bash
+kubeone install config.yaml -t . --no-init
+```
+
+## Step 4 — Restore The Backup
+
+At this point, we want to restore the backup. We'll first restore the PKI and
+then the etcd backup.
+
+{{% notice note %}}
+Tasks mentioned in this step should be run only on the leader control plane
+instance. KubeOne will automatically synchronize files with other instances.
+{{% /notice %}}
+
+First, copy the downloaded backup to the leader control plane instance.
+On Linux and macOS systems, that can be done using `rsync`, such as:
+
+```bash
+rsync -e 'ssh -J user@bastion-ip' -av ./backup user@leader-ip:~/
+```
+
+Once this is done, connect over SSH to the leader control plane instance.
+
+### Restore the etcd PKI
+
+Run the following command to restore the `etcd` PKI:
+
+```bash
+sudo mkdir -p /etc/kubernetes/pki
+sudo cp -a $HOME/backup/pki/etcd /etc/kubernetes/pki
```

Comment:
```suggestion
sudo rsync -av $HOME/backup/pki /etc/kubernetes/pki
```

Reply:
OK, here I'm definitely wrong. It should be

```bash
sudo rsync -av $HOME/backup/pki/etcd /etc/kubernetes/pki/etcd
```

Reply:
This command creates `/etc/kubernetes/pki/etcd/etcd` instead of just `/etc/kubernetes/pki/etcd`. The following one seems to work as expected:

```
sudo rsync -av $HOME/backup/pki/etcd /etc/kubernetes/pki/
```